File: src/game/entity/bomb/EnemyBomb.java
package game.entity.bomb;

import game.entity.Player;
import game.entity.blocks.DestroyableBlock;
import game.entity.models.BomberEntity;
import game.entity.models.Enemy;
import game.entity.models.Entity;

import java.util.Arrays;
import java.util.List;

public class EnemyBomb extends Bomb{
    public EnemyBomb(BomberEntity entity) {
        super(entity);
    }
    @Override
    public List<Class<? extends Entity>> getExplosionInteractionEntities() {
        return Arrays.asList(DestroyableBlock.class, Player.class, Bomb.class);
    }
}

File: src/game/entity/bomb/EnemyBomb.kt
package game.entity.bomb

import game.entity.player.Player
import game.entity.blocks.DestroyableBlock
import game.entity.player.BomberEntity
import game.entity.models.Entity

class EnemyBomb(entity: BomberEntity) : Bomb(entity) {
    override val explosionInteractionEntities: Set<Class<out Entity>>
        get() {
            return setOf(DestroyableBlock::class.java, Player::class.java, Bomb::class.java)
        }
}
